src/main/java/seedu/duke/TaskList.java

```diff
@@ -0,0 +1,154 @@
+     * @return task list.
+     */
+    public ArrayList<Task> getTaskList() {
+        return this.taskList;
+    }
+
+
+    /**
+     * Adds Task to this.tasklist.
+     * @param taskType Either a "todo", "deadline" or "event"
+     * @param input User input
+     * @throws DukeException
+     */
+    public static void addTask(String taskType, String input) throws DukeException {
+
+        switch (taskType) {
+            case "todo":
+                String[] removeTaskType = input.split("todo ");
+                String description = String.join("", removeTaskType);
+                if (description.equals("todo")) {
+                    throw new DukeException("");
+                }
+                Task todo = new ToDo(description);
+                taskList.add(todo);
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + todo);
+                break;
+            case "deadline":
+                String[] removeTaskType2 = input.split("deadline ");
+                String desAndBy = String.join("", removeTaskType2);
+                String[] sliceByDesAndBy = desAndBy.split(" /by ");
+                String description2 = sliceByDesAndBy[0];
+                String dueDateAndTime = sliceByDesAndBy[1];
+                String[] dateAndTime = dueDateAndTime.split(" ");
+                System.out.println(dateAndTime.length);
+                if (dateAndTime.length == 2) {
+                    String dueDate = dateAndTime[0];
+                    String dueTime = dateAndTime[1];
+                    LocalDate localDate = Storage.getLocalDate(dueDate);
+                    Task deadline = new Deadline(description2, localDate, dueTime);
+                    taskList.add(deadline);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + deadline);
+                } else {
+                    String dueDate = dateAndTime[0];
+                    LocalDate localDate = Storage.getLocalDate(dueDate);
+                    Task deadline = new Deadline(description2, localDate);
+                    taskList.add(deadline);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + deadline);
+                }
+                break;
+            case "event":
+                String[] removeTaskType3 = input.split("event ");
+                String desAndBy2 = String.join("", removeTaskType3);
+                String[] sliceByDesAndBy2 = desAndBy2.split(" /at ");
+                String description3 = sliceByDesAndBy2[0];
+                String dueTime2 = sliceByDesAndBy2[1];
+                Task event = new Event(description3, dueTime2);
+                taskList.add(event);
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + event);
+                break;
+
+        }

```

Comment:
Could you double check on the indentation for this again? If I remember correctly, `case` should be aligned with `switch`, though I could be wrong as well.